pyTigerGraph/pyTigerGraphBase.py

```diff
@@ -200,8 +216,52 @@ def __init__(self, host: str = "http://127.0.0.1", graphname: str = "MyGraph",
             self.awsIamHeaders["X-Amz-Security-Token"] = request.headers["X-Amz-Security-Token"]
             self.awsIamHeaders["Authorization"] = request.headers["Authorization"]
 
+        if self.jwtToken:
+            self._verify_jwt_token_support()
+
         logger.info("exit: __init__")
 
+    def _set_auth_header(self):
+        """Set the authentication header based on available tokens or credentials."""
+        if self.jwtToken:
+            return {"Authorization": "Bearer " + self.jwtToken}
+        elif self.apiToken:
+            return {"Authorization": "Bearer " + self.apiToken}
+        else:
+            return {"Authorization": "Basic {0}".format(self.base64_credential)}
+
+    def _verify_jwt_token_support(self):
+        try:
+            logger.debug("Attempting to verify JWT token support with getVer() on RestPP server.")
+            self.getVer()
+        except requests.exceptions.HTTPError as e:
+            if e.response.status_code == 403:
+                logger.error(f"Unauthorized error: {e}. The JWT token might be invalid or expired.")
+            else:
+                logger.error(f"HTTP error occurred: {e}")
+            # logger.error("The DB version using doesn't support JWT token for RestPP. Please switch to API token or username/password.")
+                raise
+        except Exception as e:
+            logger.error(f"Error occurred: {e}. The DB version using doesn't support JWT token for RestPP.")
+            logger.error("Please switch to API token or username/password.")
+            raise
+
+        # Check JWT support for GSQL server
+        try:
+            logger.debug(f"Attempting to get schema with URL: {self.gsUrl + '/gsqlserver/gsql/schema?graph=' + self.graphname}")
+            logger.debug(f"Using auth header: {self.authHeader}") 
+            self._get(self.gsUrl + "/gsqlserver/gsql/schema?graph=" + self.graphname, authMode="token")
```

Comment:
The graphname is a required field accoroding to https://docs.tigergraph.com/tigergraph-server/current/api/built-in-endpoints#_show_graph_schema_metadata. We don't have a package to test on the GSQL auth yet. So if there no graph in the database, it could cause issue. might need to change to another function for validation after testing on.